cns/requestcontroller/kubernetes/nodenetworkconfigreconciler.go

```diff
@@ -0,0 +1,54 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NodeNetworkConfigReconciler (aka controller) watches API server for any creation/deletion/updates of NodeNetworkConfig objects
+type NodeNetworkConfigReconciler struct {
+	K8sClient   K8sClient
+	RestService *restserver.HTTPRestService
+	HostName    string
+}
+
+// Reconcile relays status changes in NodeNetworkConfig to CNS
+// Returning non-nil error causes a requeue
+// Returning ctrl.Result{}, nil causes the queue to "forget" the item
+// Other return values are possible, see kubebuilder docs for details
+func (n *NodeNetworkConfigReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// We are only interested in requests coming for the node that this program is running on
+	// Requeue if it's not for this node
+	if request.Name != n.HostName {
+		return reconcile.Result{}, errors.New("Requeing")
```

Comment:
we don't actually want to requeue here.

Reply:
Removed it altogether by filtering for this in eventfilter.go